src/C++/Session.cpp

```diff
@@ -778,13 +800,11 @@ void Session::generateHeartbeat( const Message& testRequest )
 
   heartbeat.getHeader().setField( MsgType( "0" ) );
   fill( heartbeat.getHeader() );
-  try
-  {
-    TestReqID testReqID;
-    testRequest.getField( testReqID );
+
+  TestReqID testReqID;
+  if(testRequest.getFieldIfSet( testReqID )){
     heartbeat.setField( testReqID );
```

Comment:
instead of throwing an exception and catching it, simply changed to check if field exists and setting the value if it does exist